File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.py
from collections import deque

n,m = map(int, input().split())
tc = list(map(int, input().split()))
queue = deque([i for i in range(1, n+1)])

count = 0
for i in tc:
    while True:
        if queue[0] == i:
            queue.popleft()
            break
        else:
            if queue.index(i) < len(queue)/2:
                queue.append(queue.popleft())
                count += 1
            else:
                queue.appendleft(queue.pop())
                count += 1
print(count)

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*
import kotlin.collections.ArrayDeque

fun main() {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(reader.readLine())
    val n = st.nextToken().toInt()
    val m = st.nextToken().toInt()
    val numbers = StringTokenizer(reader.readLine()).let {
        IntArray(m) { _ -> it.nextToken().toInt() }
    }

    val deque = ArrayDeque<Int>()
    for (i in 1..n) {
        deque.addLast(i)
    }

    var count = 0
    for (num in numbers) {
        val index = deque.indexOf(num)
        if (index <= deque.size / 2) {
            repeat(index) {
                deque.addLast(deque.removeFirst())
                count++
            }
        } else {
            repeat(deque.size - index) {
                deque.addFirst(deque.removeLast())
                count++
            }
        }
        deque.removeFirst()
    }
    println(count)
}